Documentation/hubble.rst

```diff
@@ -0,0 +1,99 @@
+****************
+Hubble internals
+****************
+
+.. note:: This documentation section is targeted at developers who are
+          interested in contributing to Hubble. For this purpose, it describes
+          Hubble internals.
+
+.. note:: This documentation covers the Hubble server (sometimes referred as
+          "Hubble embedded") and Hubble-relay components but does not cover the
+          Hubble UI and CLI.
+
+Hubble builds on top of Cilium and eBPF to enable deep visibility into the
+communication and behavior of services as well as the networking infrastructure
+in a completely transparent manner. One of the design goals of Hubble is to
+achieve all of this at large scale. Although the initial release of Hubble was
+a standalone component, as of Cilium 1.8, Hubble server is embedded into the
+Cilium agent. This decision has been taken due to performance concerns and
+drift with the Hubble standalone approach as Hubble had to replicate a lot of
+the Cilium state in its process by calling Cilium API. This not only
+contributed to performance overhead but more importantly was the source of a
+lot of code duplication and maintenance burden.
```

Comment:
This paragraph is written from the perspective of someone who knows what Hubble did in the past and compares with the older version. I don't think this makes sense.

IMO we should state the goals we are trying to achieve and how the architecture achieves those goals. References to older standalone architecture will date quickly and just add unnecessary detail to new contributors.

Reply:
In particular from `Although the initial release of Hubble was a standalone component, as of Cilium 1.8` onwards.